### PR TITLE
Add target selection for cargo kani

### PR DESCRIPTION
**Pull request by celinval**
### Description of changes: 

Add the following options to restrict `cargo kani` verification target:
- `--bin <BIN_TARGET>`
- `--bins`
- `--lib`.

I found this very helpful for example to skip integration tests when running `cargo kani` with `--tests`.

### Resolved issues:

N/A

### Related RFC:

Optional #ISSUE-NUMBER.

### Call-outs:

1. Given the driver architecture, I explicitly added a check to ensure standalone Kani does not accept the new arguments.
2. These options were already added to `cargo kani playback` in #2491.

### Testing:

* How is this change tested? New tests

* Is this a refactor change? No

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Comment by zhassan-aws** [comment]
> In this change, we add those options to the overall verification workflow.

Can you clarify a bit the purpose of this change? Is it a cleanup PR or is it adding a new feature?

**Comment by celinval** [comment]
> > In this change, we add those options to the overall verification workflow.
> 
> Can you clarify a bit the purpose of this change? Is it a cleanup PR or is it adding a new feature?

Sure. We are adding a new feature, and I believe the title "Add target selection" reflects that, but I can definitely expand the description.

File: kani-driver/src/args/cargo.rs [hunk 1]
```diff
@@ -94,9 +94,9 @@ impl ValidateArgs for CargoCommonArgs {
     }
 }
 
-/// Arguments that Kani pass down into Cargo test essentially uninterpreted.
+/// Arguments that cargo Kani supports to select build / test target.
```
Comment (zhassan-aws) [inline kani-driver/src/args/cargo.rs:97]: Adding a link to https://doc.rust-lang.org/cargo/commands/cargo-test.html#target-selection to clarify what is meant by `target` could be useful.

File: kani-driver/src/args/mod.rs [hunk 2]
```diff
@@ -430,9 +435,38 @@ impl CheckArgs {
...
+        check_no_cargo_opt(!self.verify_opts.target.bin.is_empty(), "--bin")?;
+        // Cargo common arguments.
+        check_no_cargo_opt(self.verify_opts.cargo.all_features, "--all-features")?;
+        check_no_cargo_opt(self.verify_opts.cargo.no_default_features, "--no-default-features")?;
+        check_no_cargo_opt(!self.verify_opts.cargo.features().is_empty(), "--features / -F")?;
+        check_no_cargo_opt(!self.verify_opts.cargo.package.is_empty(), "--package / -p")?;
+        check_no_cargo_opt(!self.verify_opts.cargo.exclude.is_empty(), "--exclude")?;
+        check_no_cargo_opt(self.verify_opts.cargo.workspace, "--workspace")?;
+        check_no_cargo_opt(self.verify_opts.cargo.manifest_path.is_some(), "--manifest-path")?;
+        check_no_cargo_opt(self.verify_opts.cargo.workspace, "--workspace")?;
```
Comment (zhassan-aws) [inline kani-driver/src/args/mod.rs:469]: This is repeated two lines above

**Review by zhassan-aws — APPROVED** [review]
Nice work! The tests are neat.

File: kani-driver/src/args/mod.rs (hunk 2 above)
Comment (celinval) [reply kani-driver/src/args/mod.rs:469]: oops

**Pull request merged**